src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int MY_DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size = 0;
+    private int threshold;
+    private Node<K, V> [] table;
+
+    public MyHashMap() {
+        this.table = new Node[MY_DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (MY_DEFAULT_INITIAL_CAPACITY * LOAD_FACTOR);
+    }
+
+    class Node<K, V> {
+        private final int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key) % table.length;
+        Node<K, V> current = table[index];
+        if (current == null) {
+            table[index] = new Node<>(hash(key), key, value, null);
+            size++;
+            checkResize();
+            return;
+        }
+        while (current != null) {
+            if (current.key != null && current.key.equals(key)) {
+                current.value = value;
+                return;
+            } else if (current.key == null && key == null) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                break;
+            }
+            current = current.next;
+        }
+        current.next = new Node<>(hash(key), key, value, null);
```

Comment:
Common mistakes (jv-hash-map): Diamond operator. Be attentive with the diamond operator in java. What is the difference between: `new Node<>(...)` and `new Node<K,V>(...)`and `new Node(...)`. Read more about it [here](https://www.baeldung.com/java-diamond-operator)